odbc_fdw.c

```diff
@@ -734,175 +713,169 @@ odbcGetTableSize(char *svr_dsn, char *svr_driver, char *svr_host, char *svr_port
 static void
 odbcGetQual(Node *node, TupleDesc tupdesc, List *col_mapping_list, char **key, char **value, bool *pushdown)
 {
-    ListCell *col_mapping;
-    *key = NULL;
-    *value = NULL;
-    *pushdown = false;
-#ifdef DEBUG
-    elog(NOTICE, "odbcGetQual");
-#endif
+	ListCell *col_mapping;
+	*key = NULL;
+	*value = NULL;
+	*pushdown = false;
+
+	#ifdef DEBUG
+		elog(DEBUG1, "odbcGetQual");
+	#endif
+
     if (!node)
-        return;
+		return;
 
-    if (IsA(node, OpExpr))
-    {
-        OpExpr	*op = (OpExpr *) node;
-        Node	*left, *right;
-        Index	varattno;
+	if (IsA(node, OpExpr))
+	{
+		OpExpr  *op = (OpExpr *) node;
+		Node    *left, *right;
+		Index   varattno;
 
-        if (list_length(op->args) != 2)
-            return;
+		if (list_length(op->args) != 2)
+			return;
 
-        left = list_nth(op->args, 0);
+		left = list_nth(op->args, 0);
+		if (!IsA(left, Var))
 
-        if (!IsA(left, Var))
-            return;
+			return;
 
-        varattno = ((Var *) left)->varattno;
+		varattno = ((Var *) left)->varattno;
 
-        right = list_nth(op->args, 1);
+		right = list_nth(op->args, 1);
+
+		if (IsA(right, Const))
+		{
+			StringInfoData  buf;
+			initStringInfo(&buf);
+			/* And get the column and value... */
+			*key = NameStr(tupdesc->attrs[varattno - 1]->attname);
 
-        if (IsA(right, Const))
-        {
-            StringInfoData  buf;
-            initStringInfo(&buf);
-            /* And get the column and value... */
-            *key = NameStr(tupdesc->attrs[varattno - 1]->attname);
-#ifdef DEBUG
-            elog(NOTICE, "constant type: %u", ((Const *) right)->consttype);
-#endif
             if (((Const *) right)->consttype == PROCID_TEXTCONST)
-                *value = TextDatumGetCString(((Const *) right)->constvalue);
-            else
-            {
-                return;
-            }
-
-            /* convert qual keys to mapped couchdb attribute name */
-            foreach(col_mapping, col_mapping_list)
-            {
-                DefElem *def = (DefElem *) lfirst(col_mapping);
-                if (strcmp(def->defname, *key) == 0)
-                {
-                    *key = defGetString(def);
-                    break;
-                }
-            }
-
-            /*
-             * We can push down this qual if:
-             * - The operatory is TEXTEQ
-             * - The qual is on the _id column (in addition, _rev column can be also valid)
-             */
-
-            if (op->opfuncid == PROCID_TEXTEQ)
-                *pushdown = true;
-#ifdef DEBUG
-            elog(NOTICE, "Got qual %s = %s", *key, *value);
```

Comment:
This trace is lost.

BTW, this code refers to push down, which at first does not apply to our use case.